examples/swapi/pkg/api/people/get_by_id.go

```diff
@@ -0,0 +1,42 @@
+package people
+
+import (
+	"context"
+	"github.com/caicloud/nirvana/examples/swapi/pkg/model"
+	"sort"
+	"github.com/caicloud/nirvana/definition"
+	"reflect"
+	"fmt"
+)
+
+func getByIdDefinition(people []model.Person) definition.Definition {
+	return definition.Definition{
+		Method: definition.Get,
+		Function: func(ctx context.Context, id int) (model.Person, error) {
+			index := sort.Search(len(people), func(i int) bool {
+				return people[i].Id > model.Identity(id)
+			})
+			if index < len(people) && people[index].Id == model.Identity(id) {
+				return people[index], nil
+			}
+			return model.Person{}, fmt.Errorf("not found")
```

Comment:
Should use nirvana/errors here.

Reply:
see https://github.com/caicloud/nirvana/issues/84